blocks/pdf-viewer/pdf-viewer.js

```diff
@@ -0,0 +1,143 @@
+import { fetchPlaceholders, loadScript } from '../../scripts/lib-franklin.js';
+
+/**
+ * https://developer.adobe.com/document-services/docs/overview/pdf-embed-api/gettingstarted/
+ */
+const ADOBE_DC_VIEW_SDK_SRC = 'https://documentservices.adobe.com/view-sdk/viewer.js';
+const ADOBE_DC_VIEW_SDK_READY_EVENT = 'adobe_dc_view_sdk.ready';
+const FRANKLIN_DELAYED_COMPLETED_EVENT = 'franklin.delayed_completed';
+
+const siteConfig = (await fetchPlaceholders()).config || {};
+
+let sdkLoaded = false;
+
+/**
+ * Obtain API keys here: https://documentservices.adobe.com/dc-integration-creation-app-cdn/main.html?api=pdf-embed-api#
+ *
+ * A key is domain-specific and convers all subdomains. As such the placeholders need to
+ * feature a key per <domain>.<tld>. In this case a key each for:
+ * - localhost
+ * - hlx.page
+ * - hlx.live
+ * - mammotome.com
+ *
+ * This method checks the host from `window.location` and returns the respective placeholder
+ * holding the api key.
+ *
+ * @returns {undefined|string} The key or undefined if no matching domain was found.
+ */
+const getApiKey = () => {
+  const { host } = window.location;
+
+  if (host.startsWith('localhost')) {
+    return siteConfig.pdfApiKeyLocalhost;
+  }
+
+  if (host.endsWith('.page')) {
+    return siteConfig.pdfApiKeyPage;
+  }
+
+  if (host.endsWith('.live')) {
+    return siteConfig.pdfApiKeyLive;
+  }
+
+  if (host.endsWith('mammotome.com')) {
+    return siteConfig.pdfApiKeyProduction;
+  }
+
+  return undefined;
+};
+
+// eslint-disable-next-line no-unused-vars
+const createAdobeDCViewSDKReadyHandler = (config) => (event) => {
+  // eslint-disable-next-line no-undef
+  const adobeDCView = new AdobeDC.View({
+    clientId: config.apiKey,
+    divId: config.divId,
+  });
+  adobeDCView.previewFile(
+    config.viewer.preview,
+    config.viewer.options,
+  );
+};
+
+const onEmbedPDFScriptLoaded = () => {
+  sdkLoaded = true;
+};
+
+const loadAdobeDCViewSDK = () => {
+  if (!sdkLoaded) {
+    loadScript(ADOBE_DC_VIEW_SDK_SRC, onEmbedPDFScriptLoaded);
+  }
+};
+
+const addEventListeners = (config) => {
+  document.addEventListener(
+    ADOBE_DC_VIEW_SDK_READY_EVENT,
+    createAdobeDCViewSDKReadyHandler(config),
+  );
+  document.addEventListener(
```

Comment:
currently viewer is loaded as part of delayed. we may need to add better styling / placeholder or even change behavior on-click as a followup.